src/main/java/io/quarkus/gizmo2/creator/BlockCreator.java

```diff
@@ -2158,6 +2159,22 @@ default Expr invokeDynamic(DynamicCallSiteDesc callSiteDesc, Expr... args) {
      */
     void ifElse(Expr cond, Consumer<BlockCreator> whenTrue, Consumer<BlockCreator> whenFalse);
 
+    /**
+     * An {@code if (obj == null)} conditional.
+     *
+     * @param obj the object reference to test (must not be {@code null})
+     * @param whenTrue the builder for a block to execute if the object reference is null (must not be {@code null})
+     */
+    void ifNull(Expr obj, Consumer<BlockCreator> whenTrue);
```

Comment:
Why is this not a default method that calls `if(eq(obj, Constant.ofNull(obj.type())), whenTrue)`?

Reply:
You're right, I missed the `CodeBuilder::ifnull` part in `If.Kind.EQ`. Will rewrite this to a default method.